WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsUnknownViewModel.swift

```diff
@@ -51,29 +55,93 @@ final class CardReaderSettingsUnknownViewModel: CardReaderSettingsPresentedViewM
         ServiceLocator.stores.dispatch(connectedAction)
     }
 
+    private func updateProperties() {
+        guard let foundReader = foundReader else {
+            foundReaderSerialNumber = nil
+            return
+        }
+
+        foundReaderSerialNumber = foundReader.serial
+    }
+
     /// Dispatch a request to start reader discovery
     ///
     func startReaderDiscovery() {
         discoveryState = .searching
+        updateProperties()
         didUpdate?()
 
-        let action = CardPresentPaymentAction.startCardReaderDiscovery(siteID: siteID) { _ in
-            // TODO prompt as readers become found in 4055
+        let action = CardPresentPaymentAction.startCardReaderDiscovery(siteID: siteID) { [weak self] cardReaders in
+            self?.didDiscoverReaders(cardReaders: cardReaders)
         }
         ServiceLocator.stores.dispatch(action)
     }
 
+    /// Alert the user we have a found reader
+    ///
+    func didDiscoverReaders(cardReaders: [CardReader]) {
+        /// If we are already presenting a foundReader alert to the user, ignore the found reader
+        guard discoveryState != .foundReader else {
```

Comment:
We could do these two in just one guard:

```
guard discoveryState != .fundReader, cardReaders.count > 0 else ...
```

That might or might not be more readable though. I don't have any strong feelings one way or another 🤷🏼

Reply:
I'd like to keep this one as is, since the cases I am guarding against are somewhat different